runtime/activations/activations.go

```diff
@@ -73,71 +92,58 @@ func (a Activation) ForEach(f func(string, interface{}) error) error {
 // that share data with their parents.
 //
 type Activations struct {
-	activations []Activation
+	activations []*Activation
 }
 
-// current returns the current / most nested activation,
+// Current returns the current / most nested activation,
 // which can be found at the top of the stack.
 // It returns nil if there is no active activation.
 //
-func (a *Activations) current() *Activation {
+func (a *Activations) Current() *Activation {
 	count := len(a.activations)
 	if count < 1 {
 		return nil
 	}
-	current := a.activations[count-1]
-	return &current
+	return a.activations[count-1]
 }
 
 // Find returns the value for a given key in the current activation.
 // It returns nil if no value is found
 // or if there is no current activation.
 //
 func (a *Activations) Find(key string) interface{} {
-	current := a.current()
+	current := a.Current()
 	if current == nil {
 		return nil
 	}
 	return current.Find(key)
 }
 
-// Set adds the new key value pair to the current activation.
-// The current activation is updated in an immutable way.
+// Set sets the key value pair int the current scope.
 //
 func (a *Activations) Set(name string, value interface{}) {
-	current := a.current()
+	current := a.Current()
 	// create the first scope if there is no scope
 	if current == nil {
-		a.PushCurrent()
-		current = &a.activations[0]
+		current = NewActivation(nil)
+		a.Push(current)
```

Comment:
Not sure this is an improvement, but I would have put `PushNew()` here.

Reply:
Refactored in acf5410